book/src/nodefault.md

```diff
@@ -0,0 +1,48 @@
+# Preventing the Derivation of `Default`
+
+`bindgen` will attempt to derive/impl the `Default` traits on a best-effort basis.
+Sometimes, we need customize the implement of `Default` for certain types,
+In these cases, the `nodefault` annotation can be used to prevent bindgen 
+to autoderive the `Default` traits for a type.
+
+### Library
+
+* [`bindgen::Builder::no_default`](https://docs.rs/bindgen/latest/bindgen/struct.Builder.html#method.no_default)
+
+### Command Line
+
+* `--no-default <regex>`
+
+### Annotations
+
+```c
+/**
+ * We need to specify some preset values as the Default of Header.
+ *
+ * for example:
+ *
+ * <div rustbindgen nodefault></div>
+ */
+struct Header {
+    unsigned int magic;
+    unsigned char data[252];
+};
+
+...
+```
+
+### Customize Implements
+
+```rust
```

Comment:
it seems the docs build is failing here, see:

```
---- /tmp/mdbook-hKahHa/nodefault.md - Preventing_the_Derivation_of_::_::Customize_Implements (line 36) stdout ----
error: environment variable `OUT_DIR` not defined
 --> /tmp/mdbook-hKahHa/nodefault.md:38:18
  |
4 | include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
  |                  ^^^^^^^^^^^^^^^
  |
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
error[E0412]: cannot find type `Header` in this scope
 --> /tmp/mdbook-hKahHa/nodefault.md:40:18
  |
6 | impl Default for Header {
  |                  ^^^^^^ not found in this scope
error[E0071]: expected struct, variant or union type, found type error
 --> /tmp/mdbook-hKahHa/nodefault.md:42:9
  |
8 |         Self {
  |         ^^^^ not a struct
```

Reply:
You might want to `rust,ignore` on it or such.

Reply:
Thanks for your review, it's fixed now.